pkgs/os-specific/linux/oddjob/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, pkgs, stdenv, libxml2, dbus, pam, systemd, pkg-config, autoreconfHook }:
+stdenv.mkDerivation rec {
+
+  pname = "oddjob";
+  version = "0.34.7";
+
+  src = pkgs.fetchurl {
+     url =
+       "https://pagure.io/oddjob/archive/${pname}-${version}/oddjob-${pname}-${version}.tar.gz";
+     sha256 = "sha256-SUOsMH55HtEsk5rX0CXK0apDObTj738FGOaL5xZRnIM=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs =
+    [ libxml2 dbus pam systemd ];
+
+  postPatch = ''
+    substituteInPlace configure.ac --replace 'SYSTEMDSYSTEMUNITDIR=`pkg-config --variable=systemdsystemunitdir systemd 2> /dev/null`' "SYSTEMDSYSTEMUNITDIR=${placeholder "out"}"
+    substituteInPlace configure.ac --replace 'SYSTEMDSYSTEMUNITDIR=`pkg-config --variable=systemdsystemunitdir systemd`' "SYSTEMDSYSTEMUNITDIR=${placeholder "out"}"
+  '';
+
+  configureFlags = [
+    "--prefix=${placeholder "out"}"
+    "--sysconfdir=${placeholder "out"}/etc"
+    "--with-selinux-acls=no"
+    "--with-selinux-labels=no"
+    "--disable-systemd"
```

Comment:
Why not enable systemd?

Reply:
Im doing it so that we can toggle it on the options.oddjob section. Also for preventing potential pathing problems.

Reply:
Toggle it where? Preventing what kind of problems?

Reply:
Toggle it in nixos options. I thought by disabling the generated service I can have full control over it in Nix and can set the correct path to the executable and have it tied to the `options.services.oddjobd.enable` option.

Reply:
Either we patch the generated service file or only replace parts of it which nix does automatically by creating a drop in. upstream does not have to much in their service file https://pagure.io/oddjob/blob/master/f/scripts/oddjobd.service.in and half of it is either useless or wrong with nixos.